src/main/java/Parser.java

```diff
@@ -41,10 +41,16 @@ protected String getLine() {
 		Scanner lineScanner = new Scanner(line);
 		String[] command = new String[MAX_WORDS_PER_INSTRUCTION];
 		int i = 0;
-		while(lineScanner.hasNext()) {
-			command[i++] = lineScanner.next();
+		try {
+			while (lineScanner.hasNext()) {
+				command[i++] = lineScanner.next();
+			}
+			return command;
+		} catch (ArrayIndexOutOfBoundsException e){
+			String[] message = new String[1];
```

Comment:
I don't really like this quick solution. I think that if we are going to do input checking, we shouldn't still wrap it as an Instruction. Instead Throw the error all the way back to the PlayerThread and let the PlayerThread handle this. Thje PlayerThread should then realize this is obviously malformed and the Playerthread has the authority to send it back